internal/host/validator.go

```diff
@@ -1626,3 +1634,35 @@ func (v *validator) noSkipMissingDisk(c *validationContext) (ValidationStatus, s
 	}
 	return ValidationSuccess, successMessage
 }
+
+func (v *validator) serviceCanConnectToSpokeKubeAPI(c *validationContext) (ValidationStatus, string) {
+	// When adding a day2 host to a spoke cluster and the service operates in kube-API mode,
+	// the service needs to access the spoke cluster's kube API in order to perform some operations,
+	// such as listing nodes and approving CSRs (depending on platform).
+	// This validation makes sure in advance that the service is authorized to do that.
+	if c.cluster == nil || !c.kubeApiEnabled || *c.cluster.Status != models.ClusterStatusAddingHosts {
+		return ValidationSuccessSuppressOutput, "Host is not part of a Day2 cluster, no need to check KubeAPI connectivity"
+	}
+
+	spokeK8sClientFactory := spoke_k8s_client.NewSpokeK8sClientFactory(v.log)
+	client, err := spokeK8sClientFactory.CreateFromStorageKubeconfig(context.TODO(), c.host.ClusterID, c.objectHandler)
+	if err != nil {
+		return ValidationFailure, "Could not create the spoke k8s client connection using kubeconfig"
+	}
+
+	for _, check := range [][]string{
+		{"list", "CertificateSigningRequests"},
+		{"approve", "CertificateSigningRequests"},
+		{"list", "Nodes"},
+	} {
+		state, err := client.IsActionPermitted(check[0], check[1])
```

Comment:
Some aliases could make the rest of the code nicer to read and emphasize the intention / meaning of those strings

```suggestion
		verb, resource := check[0], check[1]
		state, err := client.IsActionPermitted(verb, resource)
```